OpenRA.Game/Campaigns.cs

```diff
@@ -0,0 +1,130 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace OpenRA
+{
+	public class ProgressionSettings
+	{
+		[Desc("List of IDs of successfully finished missions.")]
+		public string FinishedMissions = "none";
+
+		[Desc("List of special mission flags that have been triggered.")]
+		public string TriggeredFlags = "none";
+	}
+
+	public class Campaigns
```

Comment:
Can you please generalize this into a general key-value (<string, string>) store like I mentioned in #7213?  We can then expose a Lua API (to singleplayer missions only) to store not just campaign progression, but also e.g. scores (which can be shown in the mission browser), carry-over cash, the airfield/other actor killed flag for that mission in TD, base layout for that mission in RA, and so on.  We can leave it up to the individual mission scripts to serialize these various things to string and back.